src/components/structures/UserSettings.js

```diff
@@ -101,6 +108,7 @@ module.exports = React.createClass({
             middleOpacity: 1.0,
         });
         dis.unregister(this.dispatcherRef);
+        MatrixClientPeg.get().removeListener("RoomMember.membership", this._onInviteStateChange);
```

Comment:
yup, `MatrixClientPeg.get()` will return null after you log out.

In general you are right, but this is in an `componentWillUnmount` method which will be called when the user logs out. You should find that any other references to `MatrixClientPeg.get` in `componentWillUnmount` have null guards - any which don't are also bugs.

(We're actually in the process of switching to passing the MatrixClient via the react context rather than using the MatrixClientPeg singleton, which means that components can rely on the MatrixClient not changing under them. But we're already using MatrixClientPeg in this component and I'd rather not mix usage in a single component.)